modules/contact/src/auxkernels/MortarFrictionalPressureVectorAux.C

```diff
@@ -0,0 +1,124 @@
+  params.addRequiredCoupledVar(
+      "tangent_two",
+      "Second tangent vector Lagrange's multiplier for computing the mortar "
+      "frictional pressure vector.");
+  params.addRequiredParam<unsigned int>("component",
+                                        "Cartesian component of frictional tangent vector");
+
+  params.addRequiredParam<BoundaryName>("primary_boundary",
+                                        "The name of the primary boundary sideset.");
+  params.addRequiredParam<BoundaryName>("secondary_boundary",
+                                        "The name of the secondary boundary sideset.");
+
+  return params;
+}
+
+MortarFrictionalPressureVectorAux::MortarFrictionalPressureVectorAux(const InputParameters & params)
+  : AuxKernel(params),
+    _tangent_one(&coupledValueLower("tangent_one")),
+    _tangent_two(&coupledValueLower("tangent_two")),
+    _fe_problem(*params.get<FEProblemBase *>("_fe_problem_base")),
+    _primary_id(_fe_problem.mesh().getBoundaryID(getParam<BoundaryName>("primary_boundary"))),
+    _secondary_id(_fe_problem.mesh().getBoundaryID(getParam<BoundaryName>("secondary_boundary"))),
+    _component(getParam<unsigned int>("component"))
+{
+  // Only consider nodal quantities
+  if (!isNodal())
+    mooseError(
+        "MortarFrictionalPressureVector auxiliary kernel can only be used with nodal kernels.");
+
+  // Kernel need to be boundary restricted
+  if (!this->_bnd)
+    paramError("boundary",
+               "MortarFrictionalPressureVector auxiliary kernel must be restricted to a boundary.");
+
+  // Get mortar interfaces
+  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
+
+  if (displaced_mortar_interfaces.size() == 0)
+    paramError("tangent_one",
+               "No mortar interface could be identified in this problem. Make sure mortar contact "
+               "is enabled.");
+
+  const auto mortar_dimension = _fe_problem.mesh().dimension() - 1;
+  if (mortar_dimension == 1)
+    paramError("tangent_two",
+               "MortarFrictionalPressureVector auxiliary kernel can only be used in "
+               "three-dimensional problems");
+}
+
+Real
+MortarFrictionalPressureVectorAux::computeValue()
+{
+  const auto & displaced_mortar_interfaces = _fe_problem.getMortarInterfaces(/*displaced=*/true);
+
+  // Get automatic generation object for the boundary pair this auxiliary acts on.
+  if (displaced_mortar_interfaces.count(std::make_pair(_primary_id, _secondary_id)) != 1)
+    mooseError("primary_boundary",
+               "The boundary pairs do not correspond to a single mortar contact boundary pair. "
+               "Please revise your input file for proper mortar contact constraints and mortar "
+               "frictional pressure vector auxiliary variable definition.");
+
+  const AutomaticMortarGeneration & mortar_generation_object =
+      displaced_mortar_interfaces.at(std::make_pair(_primary_id, _secondary_id));
```

Comment:
Is it possible to do this setup in the constructor? It looks like you are already doing most of it there ... just have to create a pointer member for the automatic mortar generation object. Also let's use `libmesh_map_find` instead of `at` when we are confident a failed find would represent a developer failure. And then if a failed find would ever be due to a user input failure we should be doing an explicit map find and then `mooseError` (which I believe you are already doing in the constructor so 👍 )